tools/simple-httpbin/src/main/java/org/enso/shttp/SimpleHTTPBin.java

```diff
@@ -0,0 +1,97 @@
+package org.enso.shttp;
+
+import com.sun.net.httpserver.HttpHandler;
+import com.sun.net.httpserver.HttpServer;
+import sun.misc.Signal;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+
+public class SimpleHTTPBin {
+
+  private final HttpServer server;
+  private final State state;
+
+  public SimpleHTTPBin(String hostname, int port) throws IOException {
+    InetSocketAddress address = new InetSocketAddress(hostname, port);
+    server = HttpServer.create(address, 0);
+    server.setExecutor(null);
+    state = new State();
+  }
+
+  public void start() {
+    server.start();
+    state.start();
+
+    try {
+      while (state.isRunning()) {
+        Thread.sleep(1000);
+      }
+    } catch (InterruptedException e) {
+      e.printStackTrace();
+    } finally {
+      server.stop(0);
+    }
+  }
+
+  public void stop() {
+    state.stop();
+  }
+
+  public void addHandler(String path, HttpHandler handler) {
+    server.createContext(path, handler);
+  }
+
+  public static void main(String[] args) {
+    if (args.length != 2) {
+      System.err.println("Usage: SimpleHTTPServer <host> <port>");
+      System.exit(1);
+    }
+
+    String host = args[0];
+    SimpleHTTPBin server = null;
+    try {
+      int port = Integer.valueOf(args[1]);
+      server = new SimpleHTTPBin(host, port);
+      server.addHandler("/get", new DummyHandler());
+      server.addHandler("/post", new DummyHandler());
+
+      final SimpleHTTPBin server1 = server;
+      Signal.handle(
+          new Signal("TERM"),
+          (Signal sig) -> {
+            System.out.println("Stopping server...");
+            server1.stop();
+          });
+      Signal.handle(
+          new Signal("INT"),
+          (Signal sig) -> {
+            System.out.println("Stopping server...");
+            server1.stop();
+          });
```

Comment:
Can't we bind this lambda to a variable and re-use it instead of writing it twice? I know it's small but let's DRY